src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -0,0 +1,322 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+import abc
+from dataclasses import dataclass
+from typing import Optional, TypeAlias
+
+import dace
+import dace.subsets as sbs
+
+from gt4py.next.common import Dimension
+from gt4py.next.iterator import ir as itir
+from gt4py.next.iterator.ir_utils import common_pattern_matcher as cpm
+from gt4py.next.program_processors.runners.dace_fieldview import (
+    gtir_to_tasklet,
+    utility as dace_fieldview_util,
+)
+from gt4py.next.program_processors.runners.dace_fieldview.sdfg_builder import SDFGBuilder
+from gt4py.next.type_system import type_specifications as ts
+
+
+# Define aliases for return types
+SDFGField: TypeAlias = tuple[dace.nodes.Node, ts.FieldType | ts.ScalarType]
+
+DIMENSION_INDEX_FMT = "i_{dim}"
+ITERATOR_INDEX_DTYPE = dace.int32  # type of iterator indexes
+
+
+@dataclass(frozen=True)
+class PrimitiveTranslator(abc.ABC):
+    @abc.abstractmethod
+    def __call__(
+        self,
+        node: itir.Node,
+        sdfg: dace.SDFG,
+        state: dace.SDFGState,
+        sdfg_builder: SDFGBuilder,
+    ) -> list[SDFGField]:  # keep only call, same interface for all primitives
+        """Creates the dataflow subgraph representing a GTIR builtin function.
+
+        This method is used by derived classes to build a specialized subgraph
+        for a specific builtin function.
+
+        Returns a list of SDFG nodes and the associated GT4Py data type.
+
+        The GT4Py data type is useful in the case of fields, because it provides
+        information on the field domain (e.g. order of dimensions, types of dimensions).
+        """
+
+
+class AsFieldOp(PrimitiveTranslator):
+    """Generates the dataflow subgraph for the `as_field_op` builtin function."""
+
+    callable_args: list[itir.Expr]
+
+    def __init__(self, node_args: list[itir.Expr]):
+        self.callable_args = node_args
+
+    def __call__(
```

Comment:
Just a question since this is a long function: would it be possible to split its different parts into smaller and reusable functions?

Reply:
Good idea. I have extracted a couple of helper methods.